packages/flutter/test/material/app_bar_test.dart

```diff
@@ -187,7 +187,23 @@ void main() {
       ),
     );
 
-    expect(tester.getTopRight(find.text('X')).dx, 800.0);
+    expect(tester.getTopRight(find.text('X')).dx, 800.0 - 16.0);
+  });
+
+  testWidgets('AppBar titleMargin:32 title start edge is 32.0 (LTR)', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      new MaterialApp(
+        home: new Scaffold(
+          appBar: new AppBar(
+            centerTitle: false,
+            titleMargin: 32.0,
+            title: const Text('X'),
+          ),
+        ),
+      ),
+    );
+
+    expect(tester.getTopLeft(find.text('X')).dx, 32.0);
```

Comment:
can you test this in RTL also?

Reply:
also it should probably verify that the width is what you expect (i.e. that the margin is applied on the left _and_ right, not just the left)

Reply:
Done.